packages/docusaurus-plugin-content-blog/src/feed.ts

```diff
@@ -79,6 +69,26 @@ async function generateBlogFeed({
       },
     } = post;
 
+    const potentialPaths = [
+      path.join(
+        outDir,
+        permalink.replace(siteConfig.baseUrl, ''),
+        'index.html',
+      ),
+      path.join(outDir, `${permalink.replace(siteConfig.baseUrl, '')}.html`),
+    ];
+
+    const HTMLPath = potentialPaths.find(fs.pathExistsSync);
```

Comment:
We should avoid blocking FS calls, later we want to enable ESLint to prevent those so let's not introduce any new one

This code in utils can serve as inspiration:

```tsx
export async function findFolderContainingFile(
  folderPaths: string[],
  relativeFilePath: string,
): Promise<string | undefined> {
  return findAsyncSequential(folderPaths, (folderPath) =>
    fs.pathExists(path.join(folderPath, relativeFilePath)),
  );
}
```